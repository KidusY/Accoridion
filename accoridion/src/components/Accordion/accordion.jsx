import React from 'react';

class Accordion extends React.Component {
	constructor (props) {
		super();
		this.state = {
			render : false
		};
	}
	setRender = () => {
		this.setState({ render: !this.state.render });
		console.log('button');
	};
	static defaultProps = {
		section : {
			title   : '',
			content : ''
		}
	};
	render () {
		return (
			<div>
				<ul>
					<li>
						{' '}
						<button onClick={() => this.setRender()} key={this.key}>
							{' '}
							{this.props.section.title}
						</button>{' '}
					</li>
					{this.state.render && (
						<li>
							<p>{this.props.section.content}</p>
						</li>
					)}
				</ul>
			</div>
		);
	}
}

export default Accordion;
