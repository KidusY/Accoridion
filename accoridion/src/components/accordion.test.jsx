import React from 'react';
import renderer from 'react-test-renderer';
import Accordion from './Accordion/accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
const sectionsProp = {
	title   : '',
	content : ''
};
describe('<Accordion />', () => {
	it('renders an empty li when the sections prop is not supplied.', () => {
		const wrapper = shallow(<Accordion />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('Renders no section active by default', () => {
		const wrapper = shallow(<Accordion sections={sectionsProp} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
	it('Opens a clicked section', () => {
		const wrapper = shallow(<Accordion sections={sectionsProp} />);
		console.log('>>> FIND(BUTTON) <<<');
		console.log(wrapper.find('button').debug());
		wrapper.find('button').at(0).simulate('click');
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
