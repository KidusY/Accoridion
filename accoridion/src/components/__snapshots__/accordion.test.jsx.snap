// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> Opens a clicked section 1`] = `
<div>
  <ul>
    <li>
       
      <button
        onClick={[Function]}
      >
         
      </button>
       
    </li>
    <li>
      <p />
    </li>
  </ul>
</div>
`;

exports[`<Accordion /> Renders no section active by default 1`] = `
<div>
  <ul>
    <li>
       
      <button
        onClick={[Function]}
      >
         
      </button>
       
    </li>
  </ul>
</div>
`;

exports[`<Accordion /> renders an empty li when the sections prop is not supplied. 1`] = `
<div>
  <ul>
    <li>
       
      <button
        onClick={[Function]}
      >
         
      </button>
       
    </li>
  </ul>
</div>
`;
